crates/fuel-core/src/database/block.rs

```diff
@@ -110,40 +76,14 @@ impl StorageMutate<FuelBlocks> for Database {
             .storage_as_mut::<FuelBlocks>()
             .insert(key, value)?;
 
-        let height = value.header().height();
-        let block_id = value.id();
-        self.storage::<FuelBlockSecondaryKeyBlockHeights>()
-            .insert(&block_id, key)?;
-
-        // Get latest metadata entry
-        let prev_metadata = self
-            .iter_all::<FuelBlockMerkleMetadata>(Some(IterDirection::Reverse))
-            .next()
-            .transpose()?
-            .map(|(_, metadata)| metadata)
-            .unwrap_or_default();
-
-        let storage = self.borrow_mut();
-        let mut tree: MerkleTree<FuelBlockMerkleData, _> =
-            MerkleTree::load(storage, prev_metadata.version())
-                .map_err(|err| StorageError::Other(anyhow::anyhow!(err)))?;
-        tree.push(block_id.as_slice())?;
-
-        // Generate new metadata for the updated tree
-        let root = tree.root();
-        let version = tree.leaves_count();
-        let metadata = DenseMerkleMetadata::new(root, version);
-        self.storage::<FuelBlockMerkleMetadata>()
-            .insert(height, &metadata)?;
-
         // TODO: Temporary solution to store the block height in the database manually here.
         //  Later it will be controlled by the `commit_changes` function on the `Database` side.
         //  https://github.com/FuelLabs/fuel-core/issues/1589
```

Comment:
is this comment still relevant? It looks like we're storing by id vs height here now.